### DISCUSSION TITLE
this.components.componentsById[componentId] is undefined

### DISCUSSION THREAD

**Discussion by loranger** [livewire/livewire#5112] [1. Help]
Hi,

I've got a weird issue I cannot reproduce on any brand new project I could bootstrap, but my actual existing project cannot use livewire with alpine flawlessly.

I have this component (created using `artisan make:livewire RegisterWizard`). Its view starts with this root div:

```html
<div x-data="{current: @entangle('count'), color: 'blue'}">
...
```

But as soon as it's rendered, my web console warns

```
Alpine Expression Error: can't access property "$wire", this.components.componentsById[componentId] is undefined
Expression: "{current: window.Livewire.find('').entangle('count'), color: 'blue'}"
<div wire:id="tCSNau2vKqPhdCHDS223" x-data="{current: window.Livewir…count'), color: 'blue'}">
```
then

```
Uncaught TypeError: can't access property "$wire", this.components.componentsById[componentId] is undefined
```

and so on…

Whene I look at the source, I could understand alpine complains because there is no id given for the find method:

```html
    <div wire:id="h7DChxRSB2sEgJGCLmOR" wire:initial-data="{&quot;fingerprint&quot;:{&quot;id&quot;:&quot;h7DChxRSB2sEgJGCLmOR&quot;,&quot;name&quot;:&quot;register-wizard&quot;,&quot;locale&quot;:&quot;fr&quot;,&quot;path&quot;:&quot;fr\/signup\/register&quot;,&quot;method&quot;:&quot;GET&quot;,&quot;v&quot;:&quot;acj&quot;},&quot;effects&quot;:{&quot;listeners&quot;:[]},&quot;serverMemo&quot;:{&quot;children&quot;:{&quot;register-steps.account-step&quot;:{&quot;id&quot;:&quot;r30gW9lKBatu1IiXRdQe&quot;,&quot;tag&quot;:&quot;div&quot;},&quot;register-steps.plan-step&quot;:{&quot;id&quot;:&quot;x9LjyZ7oyiabjdrDmgbF&quot;,&quot;tag&quot;:&quot;div&quot;},&quot;register-steps.payment-step&quot;:{&quot;id&quot;:&quot;IytncNm4SWNPdURhDRsC&quot;,&quot;tag&quot;:&quot;div&quot;}},&quot;errors&quot;:[],&quot;htmlHash&quot;:&quot;69342273&quot;,&quot;data&quot;:{&quot;key&quot;:0,&quot;current&quot;:&quot;register-steps.account-step&quot;},&quot;dataMeta&quot;:[],&quot;checksum&quot;:&quot;3b573b5d42a0b50e51f28841ce188c044bcc85da367f41c1415417b5f650596d&quot;}}"
    x-data="{current: window.Livewire.find('').entangle('count'), color: 'blue'}"
    >
```

I use laravel 9, livewire 2.10 and alpine 3.9.1

Can you tell me what I did miss ?

**Comment by joshhanley** [comment]
@loranger hmm that seems strange, it seems like the `window.livewire.find('')` isn't being populated correctly.

Can your share your full component class and blade view so we can have a look?

Also, maybe try changing `@entangle('count')` to `$wire.entangle('count')` and see if that works?

Thanks!

**Comment by AwesomeKuro** [comment]
FINALLY
after like 3 days looking... i have the exact same problem, just extended a couple of fields from the update-profile-information-form.blade.php from the jetstream package, and i got both the $wire not defined and multiple root elements messages.

Also, because the x-action-message is not getting the wire ID of the form, it doesn't show up, it gets populated with

`window.livewire.find('')`

And that seems to be the only real issue, i can copy paste the wire id from the form and get it to show on the button click!

**Comment by AwesomeKuro** [comment]
nvm, i found the answer for my problem, there are two things i noted

first, I overrided __construct method, thus stripping the wire ID from the component mid construction (parameter id for __construct didn't reach my component)
second, it's not documented officially, but aren't root divs not supposed to have x-data on them? if you need it, you can wrap it inside another `<div>...</div>` tags
